daemon/daemon.go

```diff
@@ -1451,6 +1452,15 @@ func (daemon *Daemon) verifyContainerSettings(hostConfig *containertypes.HostCon
 				return nil, err
 			}
 		}
+
+		// Validate if the given hostname is RFC 1123 (https://tools.ietf.org/html/rfc1123) compliant.
+		if len(config.Hostname) > 0 {
+			// Linux hostname is limited to HOST_NAME_MAX=64, not not including the terminating null byte.
```

Comment:
ping @jhowardmsft @darrenstahlmsft Is there such a limitation for windows ? (point is should this be there only for linux, or …)